Automattic/jetpack
# Masterbar: remove Sites, Reader, and Account submenus

Fixes #11744

Closes #9098
Closes #9082
Closes #8288
Closes #8038
Closes #7642
Closes #7011
Closes #7009

#### Changes proposed in this Pull Request:

* Internal reference: p9Jlb4-ML-p2
* Related change on WordPress.com: D25769-code

The aim of this PR is to remove all drawer menus of the Masterbar, except for the Notifications.

Today, when you click on "My Site", "Reader", or on your avatar, a drawer menu opens up with sub-menus. This PR removes all those.

I also took that opportunity to fix all PHPCS warnings in the Masterbar module, and add it to the pre-commit hook.

#### Testing instructions:

* Go to Jetpack > Settings > Writing, and enable the "WordPress.com toolbar" feature.
* After refreshing, you should see the traditional masterbar, just like before:
![image](https://user-images.githubusercontent.com/426388/55244182-f72fcb80-5240-11e9-90ee-85dd2e076e8a.png)
* However, when clicking on any of the items you should be redirected to Calypso with no errors.

You can also take that opportunity to:

- Check for JavaScript errors on the page.
- Check if Tracks events are properly triggered when clicking on each link. You should see those events in your browser console after submitting `localStorage.setItem( 'debug', 'dops:analytics' );` to enable debugging tracks.
![image](https://user-images.githubusercontent.com/426388/55244364-5988cc00-5241-11e9-9e17-e4a70c5a9e6c.png)

#### Follow-up:

- Once this is merged and shipped, and after a few versions so that we know that the majority of our userbase has updated, we should be able to update 2 things:
- `wp-content/mu-plugins/admin-bar/masterbar-overrides/masterbar.js` to remove unneeded elements.
- remove `wpcom_jetpack_masterbar_logout` on WordPress.com; it handles the sync of the log out event in the masterbar.

#### Proposed changelog entry for your changes:

* WordPress.com toolbar: simplify toolbar by removing drawer menus.
